terminal/src/Format.hs

```diff
@@ -138,57 +147,69 @@ format flags (Env inputs) =
 validate :: Env -> Task.Task Exit.Format ()
 validate (Env inputs) = do
   case inputs of
-    Stdin ->
-      do
-        original <- Task.io BS.getContents
-        case formatByteString Parse.Application original of
-          Nothing ->
-            error "TODO: report error"
-          Just formatted ->
-            when (BSL.fromStrict original /= B.toLazyByteString formatted) $
-              Task.throw Exit.FormatValidateNotCorrectlyFormatted
+    Stdin -> do
+      result <- formatByteString Parse.Application Nothing <$> Task.io BS.getContents
+      throwIfHasValidateErrors [result]
     Files paths ->
       validateFiles Parse.Application paths
     Project projectType paths ->
       validateFiles projectType paths
 
 validateFiles :: Parse.ProjectType -> [FilePath] -> Task.Task Exit.Format ()
 validateFiles projectType paths = do
-  validationResults <- mapM (validateFile projectType) paths
-  when (any (== False) validationResults) $
-    Task.throw Exit.FormatValidateNotCorrectlyFormatted
+  results <- mapM (validateFile projectType) paths
+  throwIfHasValidateErrors results
+
+throwIfHasValidateErrors :: [FormattingResult] -> Task.Task Exit.Format ()
```

Comment:
Just a minor thought: this function feels a bit overly-specific.  Maybe a more composable function would be something like
```haskell
throwIfNonEmpty :: [a] -> Task.Task (NonEmpty a) ()
```
(and maybe even belongs in the `Task` module)

and would be used above as
```haskell
Task.mapError Exit.FormatValidateErrors $
    throwIfNonEmpty (mapMaybe validateFailure results)
```

Reply:
`throwIfHasFormattingErrors` below could also be replaced similarly